File: articles/causal-inference-and-computational-reasoning/cpp/causal_effect.cpp
#include <iostream>

struct Contrast {
    double treated_mean;
    double control_mean;
    double effect() const { return treated_mean - control_mean; }
};

int main() {
    Contrast contrast{0.64, 0.47};
    std::cout << "causal contrast = " << contrast.effect() << "\n";
    return 0;
}

File: articles/causal-inference-and-computational-reasoning/java/CausalEffect.java
public class CausalEffect {
    static double effect(double treatedMean, double controlMean) {
        return treatedMean - controlMean;
    }

    public static void main(String[] args) {
        System.out.printf("causal contrast = %.4f%n", effect(0.64, 0.47));
    }
}
